### DISCUSSION TITLE
Test for `requires-recent-login` throw programmatically

### DISCUSSION THREAD

**Discussion by FMorschel** [firebase/FirebaseUI-Flutter#158] [Ideas]
Copied from https://github.com/firebase/flutterfire/issues/11730

On `firebase_auth` there is a `FirebaseAuthException` with code "`requires-recent-login`". I believe it would be useful to have a method to test if this would be thrown on trying to work with security-sensitive actions. 

For example, with `firebase_ui_auth`, when I call `EmailSignUpDialog` it throws the warning there. I would love for it to show my users that they need to reauthenticate **before** opening that dialog.

**Comment by lesnitsky** [comment]
Could you please provide more details on what's expected?

> when I call EmailSignUpDialog it throws the warning there

I'm not sure I understand what it means.

**Reply by FMorschel** [reply]
When I use `EmailSignUpDialog`, I need to wrap it in a `FirebaseUIActions` with an `AuthStateChangeAction<AuthFailed>` like the following code:

```dart
FirebaseUIActions(
  actions: [
    AuthStateChangeAction<AuthFailed>((context, state) {
      final exception = state.exception;
      if (exception is FirebaseAuthException) {
        if (exception.code == 'requires-recent-login') {
          showReauthenticateDialog(
            context: context,
            providers: linkedProviders(),
          );
        }
      }
    }),
  ],
  child: EmailSignUpDialog(
    provider: provider as EmailAuthProvider,
    action: AuthAction.link,
    showPasswordVisibilityToggle: true,
  ),
)
```

Now, as I asked above;

> I would love for it to show my users that they need to reauthenticate **before** opening that dialog.

Not by throwing inside when the user tries to link the new provider. I want to ask the person to reauthenticate just before opening the dialog. For example by executing a `FirebaseAuth.instance.needsReauthentication()` or something similar.
